packages/core-bookmarks/bookmarks.js

```diff
@@ -1,7 +1,50 @@
 import './db/helpers';
+import DataLoader from 'dataloader';
 import runMigrations from './db/schema';
+import { Bookmarks } from './db/collections';
 
 export * from './db/collections';
-export default () => {
+
+export const services = {};
+
+// eslint-disable-next-line
+export default (config) => {
   runMigrations();
+  // eslint-disable-next-line
+  return (req) => {
+    const bookmarksByQueryLoader = new DataLoader(async (queries) => {
+      const results = Bookmarks.find({
+        $or: queries,
+      }).fetch();
+      return queries.map(
+        (key) =>
+          results.find(
+            (result) =>
+              result.userId === key.userId && result.productId === key.productId
+          ) || null
```

Comment:
`?? ` is probably better here, because with || you will convert every falsy value to null